tests/filecheck/dialects/riscv/riscv_ops_custom_syntax.mlir

```diff
@@ -0,0 +1,270 @@
+  %fdiv_s = "riscv.fdiv.s"(%f0, %f1) : (!riscv.freg<>, !riscv.freg<>) -> !riscv.freg<>
+  %fsqrt_s = "riscv.fsqrt.s"(%f0) : (!riscv.freg<>) -> !riscv.freg<>
+  // CHECK-NEXT: %fadd_s = riscv.fadd.s %f0, %f1 : !riscv.freg<>
+  // CHECK-NEXT: %fsub_s = riscv.fsub.s %f0, %f1 : !riscv.freg<>
+  // CHECK-NEXT: %fmul_s = riscv.fmul.s %f0, %f1 : !riscv.freg<>
+  // CHECK-NEXT: %fdiv_s = riscv.fdiv.s %f0, %f1 : !riscv.freg<>
+  // CHECK-NEXT: %fsqrt_s = riscv.fsqrt.s %f0 : !riscv.freg<>
+
+  %fsgnj_s = "riscv.fsgnj.s"(%f0, %f1) : (!riscv.freg<>, !riscv.freg<>) -> !riscv.freg<>
+  %fsgnjn_s = "riscv.fsgnjn.s"(%f0, %f1) : (!riscv.freg<>, !riscv.freg<>) -> !riscv.freg<>
+  %fsgnjx_s = "riscv.fsgnjx.s"(%f0, %f1) : (!riscv.freg<>, !riscv.freg<>) -> !riscv.freg<>
+  // CHECK-NEXT: %fsgnj_s = riscv.fsgnj.s %f0, %f1 : !riscv.freg<>
+  // CHECK-NEXT: %fsgnjn_s = riscv.fsgnjn.s %f0, %f1 : !riscv.freg<>
+  // CHECK-NEXT: %fsgnjx_s = riscv.fsgnjx.s %f0, %f1 : !riscv.freg<>
+
+  %fmin_s = "riscv.fmin.s"(%f0, %f1) : (!riscv.freg<>, !riscv.freg<>) -> !riscv.freg<>
+  %fmax_s = "riscv.fmax.s"(%f0, %f1) : (!riscv.freg<>, !riscv.freg<>) -> !riscv.freg<>
+  // CHECK-NEXT: %fmin_s = riscv.fmin.s %f0, %f1 : !riscv.freg<>
+  // CHECK-NEXT: %fmax_s = riscv.fmax.s %f0, %f1 : !riscv.freg<>
+
+  %fcvt_w_s = "riscv.fcvt.w.s"(%f0) : (!riscv.freg<>) -> !riscv.reg<>
+  %fcvt_wu_s = "riscv.fcvt.wu.s"(%f0) : (!riscv.freg<>) -> !riscv.reg<>
+  %fmv_x_w = "riscv.fmv.x.w"(%f0) : (!riscv.freg<>) -> !riscv.reg<>
+  // CHECK-NEXT: %fcvt_w_s = riscv.fcvt.w.s %f0 : !riscv.reg<>
+  // CHECK-NEXT: %fcvt_wu_s = riscv.fcvt.wu.s %f0 : !riscv.reg<>
+  // CHECK-NEXT: %fmv_x_w = riscv.fmv.x.w %f0 : !riscv.reg<>
+
+  %feq_s = "riscv.feq.s"(%f0, %f1) : (!riscv.freg<>, !riscv.freg<>) -> !riscv.reg<>
+  %flt_s = "riscv.flt.s"(%f0, %f1) : (!riscv.freg<>, !riscv.freg<>) -> !riscv.reg<>
+  %fle_s = "riscv.fle.s"(%f0, %f1) : (!riscv.freg<>, !riscv.freg<>) -> !riscv.reg<>
+  %fclass_s = "riscv.fclass.s"(%f0) : (!riscv.freg<>) -> !riscv.reg<>
+  %fcvt_s_w = "riscv.fcvt.s.w"(%0) : (!riscv.reg<>) -> !riscv.freg<>
+  %fcvt_s_wu = "riscv.fcvt.s.wu"(%0) : (!riscv.reg<>) -> !riscv.freg<>
+  %fmv_w_x = "riscv.fmv.w.x"(%0) : (!riscv.reg<>) -> !riscv.freg<>
+  // CHECK-NEXT: %feq_s = riscv.feq.s %f0, %f1 : !riscv.reg<>
+  // CHECK-NEXT: %flt_s = riscv.flt.s %f0, %f1 : !riscv.reg<>
+  // CHECK-NEXT: %fle_s = riscv.fle.s %f0, %f1 : !riscv.reg<>
+  // CHECK-NEXT: %fclass_s = riscv.fclass.s %f0 : !riscv.reg<>
+  // CHECK-NEXT: %fcvt_s_w = riscv.fcvt.s.w %0 : !riscv.freg<>
+  // CHECK-NEXT: %fcvt_s_wu = riscv.fcvt.s.wu %0 : !riscv.freg<>
+  // CHECK-NEXT: %fmv_w_x = riscv.fmv.w.x %0 : !riscv.freg<>
+
+  %flw = "riscv.flw"(%0) {"immediate" = 1 : i32}: (!riscv.reg<>) -> !riscv.freg<>
+  "riscv.fsw"(%0, %f0) {"immediate" = 1 : i32} : (!riscv.reg<>, !riscv.freg<>) -> ()
+  // CHECK-NEXT: %flw = riscv.flw %0 {"immediate" = 1 : i32} : !riscv.freg<>
+  // CHECK-NEXT: riscv.fsw %0, %f0 {"immediate" = 1 : i32} :
```

Comment:
I would like to see the immediate also be present in the custom format

Reply:
Custom syntax is implemented only for RISC-V base instructions currently. Other extensions such as F-Extension should have their own PR.